<a href="https://colab.research.google.com/github/Francisco2963/Analisis-con-Pandas-y-Kaggle-Core-/blob/main/An%C3%A1lisis_de_Pandas_y_Kaggle_Core.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Instrucciones

## Preparación del Entorno

Asegúrate de tener instalado Pandas en tu entorno de trabajo.

Descarga el archivo dataset.csv desde Kaggle. Elige un dataset que te interese y que no incluya visualización de datos. Algunas sugerencias pueden ser datasets relacionados con ventas, compras, productos, etc.

## Cargar los Datos

Carga el archivo CSV en un DataFrame de Pandas.

Muestra las primeras 10 filas del DataFrame para confirmar que los datos se han cargado correctamente.

## Exploración Inicial de los Datos

Muestra las últimas 5 filas del DataFrame.

Utiliza el método info() para obtener información general sobre el DataFrame, incluyendo el número de entradas, nombres de las columnas, tipos de datos y memoria utilizada.

Genera estadísticas descriptivas del DataFrame utilizando el método describe().

## Limpieza de Datos

Identifica y maneja los datos faltantes utilizando técnicas apropiadas (relleno con valores estadísticos, interpolación, eliminación, etc.).

Corrige los tipos de datos si es necesario (por ejemplo, convertir cadenas a fechas).

Elimina duplicados si los hay.

## Transformación de Datos

Crea nuevas columnas basadas en operaciones con las columnas existentes (por ejemplo, calcular ingresos a partir de ventas y precios).

Normaliza o estandariza columnas si es necesario.

Clasifica los datos en categorías relevantes.

## Análisis de Datos

Realiza agrupaciones de datos utilizando groupby para obtener insights
específicos (por ejemplo, ventas por producto, ventas por región, etc.).

Aplica funciones de agregación como sum, mean, count, min, max, std, y var.

Utiliza el método apply para realizar operaciones más complejas y personalizadas.

## Documentación

Documenta claramente cada paso del análisis, explicando qué se hizo y por qué se hizo.

Asegúrate de que el código sea legible y esté bien comentado.

In [100]:
import pandas as pd
import numpy as np
import matplotlib as plt

In [101]:
df = pd.read_csv('/content/titanic.csv')

In [102]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [103]:
df.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

In [104]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [105]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


# Limpieza de Datos

Identifica y maneja los datos faltantes utilizando técnicas apropiadas (relleno con valores estadísticos, interpolación, eliminación, etc.).

Corrige los tipos de datos si es necesario (por ejemplo, convertir cadenas a fechas).

Elimina duplicados si los hay.

In [106]:
df.duplicated().sum()

0

In [107]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [108]:
df.Age.info()

<class 'pandas.core.series.Series'>
RangeIndex: 891 entries, 0 to 890
Series name: Age
Non-Null Count  Dtype  
--------------  -----  
714 non-null    float64
dtypes: float64(1)
memory usage: 7.1 KB


In [109]:
df['Age'].unique()

array([22.  , 38.  , 26.  , 35.  ,   nan, 54.  ,  2.  , 27.  , 14.  ,
        4.  , 58.  , 20.  , 39.  , 55.  , 31.  , 34.  , 15.  , 28.  ,
        8.  , 19.  , 40.  , 66.  , 42.  , 21.  , 18.  ,  3.  ,  7.  ,
       49.  , 29.  , 65.  , 28.5 ,  5.  , 11.  , 45.  , 17.  , 32.  ,
       16.  , 25.  ,  0.83, 30.  , 33.  , 23.  , 24.  , 46.  , 59.  ,
       71.  , 37.  , 47.  , 14.5 , 70.5 , 32.5 , 12.  ,  9.  , 36.5 ,
       51.  , 55.5 , 40.5 , 44.  ,  1.  , 61.  , 56.  , 50.  , 36.  ,
       45.5 , 20.5 , 62.  , 41.  , 52.  , 63.  , 23.5 ,  0.92, 43.  ,
       60.  , 10.  , 64.  , 13.  , 48.  ,  0.75, 53.  , 57.  , 80.  ,
       70.  , 24.5 ,  6.  ,  0.67, 30.5 ,  0.42, 34.5 , 74.  ])

In [110]:
df['Age'].sort_values()

803    0.42
755    0.67
644    0.75
469    0.75
78     0.83
       ... 
859     NaN
863     NaN
868     NaN
878     NaN
888     NaN
Name: Age, Length: 891, dtype: float64

In [111]:
df['Age'].isnull()

0      False
1      False
2      False
3      False
4      False
       ...  
886    False
887    False
888     True
889    False
890    False
Name: Age, Length: 891, dtype: bool

In [112]:
fill_nul_age = df['Age'].isnull()
df[fill_nul_age]

PassengerId  Survived  Pclass                                      Name  \
5              6         0       3                          Moran, Mr. James   
17            18         1       2              Williams, Mr. Charles Eugene   
19            20         1       3                   Masselmani, Mrs. Fatima   
26            27         0       3                   Emir, Mr. Farred Chehab   
28            29         1       3             O'Dwyer, Miss. Ellen "Nellie"   
..           ...       ...     ...                                       ...   
859          860         0       3                          Razi, Mr. Raihed   
863          864         0       3         Sage, Miss. Dorothy Edith "Dolly"   
868          869         0       3               van Melkebeke, Mr. Philemon   
878          879         0       3                        Laleff, Mr. Kristo   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   

        Sex  Age  SibSp  Parch      Ticket     Fare Cabin Embarked  
5      male  NaN      0      0      330877   8.4583   NaN        Q  
17     male  NaN      0      0      244373  13.0000   NaN        S  
19   female  NaN      0      0        2649   7.2250   NaN        C  
26     male  NaN      0      0        2631   7.2250   NaN        C  
28   female  NaN      0      0      330959   7.8792   NaN        Q  
..      ...  ...    ...    ...         ...      ...   ...      ...  
859    male  NaN      0      0        2629   7.2292   NaN        C  
863  female  NaN      8      2    CA. 2343  69.5500   NaN        S  
868    male  NaN      0      0      345777   9.5000   NaN        S  
878    male  NaN      0      0      349217   7.8958   NaN        S  
888  female  NaN      1      2  W./C. 6607  23.4500   NaN        S  

[177 rows x 12 columns]

In [113]:
fill_nul = df.loc[15:21]
fill_nul

PassengerId  Survived  Pclass  \
15           16         1       2   
16           17         0       3   
17           18         1       2   
18           19         0       3   
19           20         1       3   
20           21         0       2   
21           22         1       2   

                                                 Name     Sex   Age  SibSp  \
15                   Hewlett, Mrs. (Mary D Kingcome)   female  55.0      0   
16                               Rice, Master. Eugene    male   2.0      4   
17                       Williams, Mr. Charles Eugene    male   NaN      0   
18  Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.0      1   
19                            Masselmani, Mrs. Fatima  female   NaN      0   
20                               Fynney, Mr. Joseph J    male  35.0      0   
21                              Beesley, Mr. Lawrence    male  34.0      0   

    Parch  Ticket    Fare Cabin Embarked  
15      0  248706  16.000   NaN        S  
16      1  382652  29.125   NaN        Q  
17      0  244373  13.000   NaN        S  
18      0  345763  18.000   NaN        S  
19      0    2649   7.225   NaN        C  
20      0  239865  26.000   NaN        S  
21      0  248698  13.000   D56        S

In [114]:
df['Age'].median()

28.0

In [115]:
df['Age'].mean()

29.69911764705882

In [116]:
df['Age'].mode()

0    24.0
Name: Age, dtype: float64

In [117]:
# Rellenar valores nulos con la media de la columna
fill_nul_1 = df['Age'] = df['Age'].fillna(df['Age'].mean())
fill_nul_1

0      22.000000
1      38.000000
2      26.000000
3      35.000000
4      35.000000
         ...    
886    27.000000
887    19.000000
888    29.699118
889    26.000000
890    32.000000
Name: Age, Length: 891, dtype: float64

In [118]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          891 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [119]:
df.Cabin.info()

<class 'pandas.core.series.Series'>
RangeIndex: 891 entries, 0 to 890
Series name: Cabin
Non-Null Count  Dtype 
--------------  ----- 
204 non-null    object
dtypes: object(1)
memory usage: 7.1+ KB


In [120]:
df['Cabin'].unique()

array([nan, 'C85', 'C123', 'E46', 'G6', 'C103', 'D56', 'A6',
       'C23 C25 C27', 'B78', 'D33', 'B30', 'C52', 'B28', 'C83', 'F33',
       'F G73', 'E31', 'A5', 'D10 D12', 'D26', 'C110', 'B58 B60', 'E101',
       'F E69', 'D47', 'B86', 'F2', 'C2', 'E33', 'B19', 'A7', 'C49', 'F4',
       'A32', 'B4', 'B80', 'A31', 'D36', 'D15', 'C93', 'C78', 'D35',
       'C87', 'B77', 'E67', 'B94', 'C125', 'C99', 'C118', 'D7', 'A19',
       'B49', 'D', 'C22 C26', 'C106', 'C65', 'E36', 'C54',
       'B57 B59 B63 B66', 'C7', 'E34', 'C32', 'B18', 'C124', 'C91', 'E40',
       'T', 'C128', 'D37', 'B35', 'E50', 'C82', 'B96 B98', 'E10', 'E44',
       'A34', 'C104', 'C111', 'C92', 'E38', 'D21', 'E12', 'E63', 'A14',
       'B37', 'C30', 'D20', 'B79', 'E25', 'D46', 'B73', 'C95', 'B38',
       'B39', 'B22', 'C86', 'C70', 'A16', 'C101', 'C68', 'A10', 'E68',
       'B41', 'A20', 'D19', 'D50', 'D9', 'A23', 'B50', 'A26', 'D48',
       'E58', 'C126', 'B71', 'B51 B53 B55', 'D49', 'B5', 'B20', 'F G63',
       'C62 C64',

In [121]:
df['Cabin'].value_counts()

Cabin
B96 B98        4
G6             4
C23 C25 C27    4
C22 C26        3
F33            3
              ..
E34            1
C7             1
C54            1
E36            1
C148           1
Name: count, Length: 147, dtype: int64

In [122]:
df['Cabin'].isnull()


0       True
1      False
2       True
3      False
4       True
       ...  
886     True
887    False
888     True
889    False
890     True
Name: Cabin, Length: 891, dtype: bool

In [123]:
fill_nul_2 = df['Cabin'].isnull()
df[fill_nul_2]

PassengerId  Survived  Pclass                                      Name  \
0              1         0       3                   Braund, Mr. Owen Harris   
2              3         1       3                    Heikkinen, Miss. Laina   
4              5         0       3                  Allen, Mr. William Henry   
5              6         0       3                          Moran, Mr. James   
7              8         0       3            Palsson, Master. Gosta Leonard   
..           ...       ...     ...                                       ...   
884          885         0       3                    Sutehall, Mr. Henry Jr   
885          886         0       3      Rice, Mrs. William (Margaret Norton)   
886          887         0       2                     Montvila, Rev. Juozas   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex        Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0      male  22.000000      1      0         A/5 21171   7.2500   NaN        S  
2    female  26.000000      0      0  STON/O2. 3101282   7.9250   NaN        S  
4      male  35.000000      0      0            373450   8.0500   NaN        S  
5      male  29.699118      0      0            330877   8.4583   NaN        Q  
7      male   2.000000      3      1            349909  21.0750   NaN        S  
..      ...        ...    ...    ...               ...      ...   ...      ...  
884    male  25.000000      0      0   SOTON/OQ 392076   7.0500   NaN        S  
885  female  39.000000      0      5            382652  29.1250   NaN        Q  
886    male  27.000000      0      0            211536  13.0000   NaN        S  
888  female  29.699118      1      2        W./C. 6607  23.4500   NaN        S  
890    male  32.000000      0      0            370376   7.7500   NaN        Q  

[687 rows x 12 columns]

In [124]:
fill_nul_2 = df.loc[885:]
fill_nul_2

PassengerId  Survived  Pclass                                      Name  \
885          886         0       3      Rice, Mrs. William (Margaret Norton)   
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex        Age  SibSp  Parch      Ticket    Fare Cabin Embarked  
885  female  39.000000      0      5      382652  29.125   NaN        Q  
886    male  27.000000      0      0      211536  13.000   NaN        S  
887  female  19.000000      0      0      112053  30.000   B42        S  
888  female  29.699118      1      2  W./C. 6607  23.450   NaN        S  
889    male  26.000000      0      0      111369  30.000  C148        C  
890    male  32.000000      0      0      370376   7.750   NaN        Q

In [125]:
df[df['Pclass'] == 1][['Name','Cabin']].sort_values(by='Name')

Name    Cabin
730                    Allen, Miss. Elisabeth Walton       B5
305                   Allison, Master. Hudson Trevor  C22 C26
297                     Allison, Miss. Helen Loraine  C22 C26
498  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  C22 C26
460                              Anderson, Mr. Harry      E12
..                                               ...      ...
155                      Williams, Mr. Charles Duane      NaN
351           Williams-Lambert, Mr. Fletcher Fellows     C128
55                                 Woolner, Mr. Hugh      C52
555                               Wright, Mr. George      NaN
325                         Young, Miss. Marie Grice      C32

[216 rows x 2 columns]

Nuestra ultima columna a explorar para NaN es la columna Cabin, que tiene 204 datos faltantes de un total de 891 filas,lo que representa un 22,89 % de datos faltantes de la columna Cabin, por lo tanto se recomienda eliminar la columna ya que no es posible establecer un metodo para poder rellenar los datos faltantes y conocer en que cabina se encontraba el pasajero

In [126]:
# Eliminar columnas con datos faltantes
df = df.drop('Cabin', axis=1)

In [127]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          891 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(4)
memory usage: 76.7+ KB


In [128]:
df.Embarked.info()

<class 'pandas.core.series.Series'>
RangeIndex: 891 entries, 0 to 890
Series name: Embarked
Non-Null Count  Dtype 
--------------  ----- 
889 non-null    object
dtypes: object(1)
memory usage: 7.1+ KB


In [129]:
df['Embarked'].unique()

array(['S', 'C', 'Q', nan], dtype=object)

In [130]:
df['Embarked'].value_counts()

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

In [131]:
df['Embarked'].isnull()

0      False
1      False
2      False
3      False
4      False
       ...  
886    False
887    False
888    False
889    False
890    False
Name: Embarked, Length: 891, dtype: bool

In [132]:
fill_nul = df['Embarked'].isnull()
df[fill_nul]

PassengerId  Survived  Pclass                                       Name  \
61            62         1       1                        Icard, Miss. Amelie   
829          830         1       1  Stone, Mrs. George Nelson (Martha Evelyn)   

        Sex   Age  SibSp  Parch  Ticket  Fare Embarked  
61   female  38.0      0      0  113572  80.0      NaN  
829  female  62.0      0      0  113572  80.0      NaN

In [133]:
fill_nul = df.loc[58:65]
fill_nul

PassengerId  Survived  Pclass                                Name     Sex  \
58           59         1       2        West, Miss. Constance Mirium  female   
59           60         0       3  Goodwin, Master. William Frederick    male   
60           61         0       3               Sirayanian, Mr. Orsen    male   
61           62         1       1                 Icard, Miss. Amelie  female   
62           63         0       1         Harris, Mr. Henry Birkhardt    male   
63           64         0       3               Skoog, Master. Harald    male   
64           65         0       1               Stewart, Mr. Albert A    male   
65           66         1       3            Moubarek, Master. Gerios    male   

          Age  SibSp  Parch      Ticket     Fare Embarked  
58   5.000000      1      2  C.A. 34651  27.7500        S  
59  11.000000      5      2     CA 2144  46.9000        S  
60  22.000000      0      0        2669   7.2292        C  
61  38.000000      0      0      113572  80.0000      NaN  
62  45.000000      1      0       36973  83.4750        S  
63   4.000000      3      2      347088  27.9000        S  
64  29.699118      0      0    PC 17605  27.7208        C  
65  29.699118      1      1        2661  15.2458        C

In [134]:
fill_nul = df.loc[827:833]
fill_nul

PassengerId  Survived  Pclass                                       Name  \
827          828         1       2                      Mallet, Master. Andre   
828          829         1       3               McCormack, Mr. Thomas Joseph   
829          830         1       1  Stone, Mrs. George Nelson (Martha Evelyn)   
830          831         1       3    Yasbeck, Mrs. Antoni (Selini Alexander)   
831          832         1       2            Richards, Master. George Sibley   
832          833         0       3                             Saad, Mr. Amin   
833          834         0       3                     Augustsson, Mr. Albert   

        Sex        Age  SibSp  Parch           Ticket     Fare Embarked  
827    male   1.000000      0      2  S.C./PARIS 2079  37.0042        C  
828    male  29.699118      0      0           367228   7.7500        Q  
829  female  62.000000      0      0           113572  80.0000      NaN  
830  female  15.000000      1      0             2659  14.4542        C  
831    male   0.830000      1      1            29106  18.7500        S  
832    male  29.699118      0      0             2671   7.2292        C  
833    male  23.000000      0      0           347468   7.8542        S

Despues de analizar la columna Embarket no es posible determinar cual es el tipo de embarque de los dos datos NaN faltantes, por lo tanto aplcaremos el relleno de nulos hacia atras

In [135]:
# Rellenar valores nulos hacia atrás
fill_nul = df['Embarked'] = df['Embarked'].bfill()
fill_nul

0      S
1      C
2      S
3      S
4      S
      ..
886    S
887    S
888    S
889    C
890    Q
Name: Embarked, Length: 891, dtype: object

In [136]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          891 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Embarked     891 non-null    object 
dtypes: float64(2), int64(5), object(4)
memory usage: 76.7+ KB


# Transformación de Datos

Crea nuevas columnas basadas en operaciones con las columnas existentes (por ejemplo, calcular ingresos a partir de ventas y precios).

Normaliza o estandariza columnas si es necesario.

Clasifica los datos en categorías relevantes.

In [137]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Embarked  
0      0         A/5 21171   7.2500        S  
1      0          PC 17599  71.2833        C  
2      0  STON/O2. 3101282   7.9250        S  
3      0            113803  53.1000        S  
4      0            373450   8.0500        S

Mostrar por pantalla el promedio de hombres y nujeres que sobrevivieron y murieron

In [138]:
df['Survived'].value_counts()

Survived
0    549
1    342
Name: count, dtype: int64

In [139]:
grouped = df.groupby('Survived')
grouped.size()

Survived
0    549
1    342
dtype: int64

edad Promedio de pasajeros hombres y mujeres que sobreviveron y murieron

In [140]:
df.groupby(['Survived', 'Sex'])['Age'].mean().round(2)

Survived  Sex   
0         female    26.02
          male      31.18
1         female    28.98
          male      27.63
Name: Age, dtype: float64

In [141]:
# Sobrevivientes por sexo
sobrevivientes_por_sexo = df.groupby('Survived')['Sex'].value_counts()
sobrevivientes_por_sexo

Survived  Sex   
0         male      468
          female     81
1         female    233
          male      109
Name: count, dtype: int64

In [142]:
# Sobrevivientes por clase
conteo_sobrevivientes_por_clase = df.groupby('Survived')['Pclass'].count()
conteo_sobrevivientes_por_clase

Survived
0    549
1    342
Name: Pclass, dtype: int64

In [143]:
# Sobrevivientes mínimas y máximas por clase y sexo
sobrevivientes_por_clase = df.groupby('Survived')['Pclass'].min()
sobrevivientes_por_sexo = df.groupby('Survived')['Sex'].max()
sobrevivientes_por_clase
sobrevivientes_por_sexo

Survived
0    male
1    male
Name: Sex, dtype: object

In [144]:
# Agrupar por 'Survived' y 'Pclass'
grouped_multi = df.groupby(['Survived', 'Pclass'])

# Suma de Age por Surived y Pclass
suma_sobrevivientes_por_clase = grouped_multi['Age'].sum()
suma_sobrevivientes_por_clase

# Promedio de Age por Survived y Pclass
promedio_sobrevivientes_por_clase = grouped_multi['Age'].mean()
promedio_sobrevivientes_por_clase


Survived  Pclass
0         1         40.896074
          2         33.266947
          3         27.417500
1         1         34.784615
          2         26.076166
          3         23.232689
Name: Age, dtype: float64

In [145]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  891.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   13.002015    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   22.000000    0.000000   
50%     446.000000    0.000000    3.000000   29.699118    0.000000   
75%     668.500000    1.000000    3.000000   35.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [146]:
# Desviación estándar de los sobrevivientes por clase
desviacion_estandar_sobrevivientes = df.groupby('Pclass')['Survived'].std()
desviacion_estandar_sobrevivientes

Pclass
1    0.484026
2    0.500623
3    0.428949
Name: Survived, dtype: float64